mailpoet/lib/Newsletter/NewsletterValidator.php

```diff
@@ -0,0 +1,104 @@
+  private $subscribersFeature;
+
+  public function __construct(
+    Bridge $bridge,
+    TrackingConfig $trackingConfig,
+    SubscribersFeature $subscribersFeature
+  ) {
+    $this->bridge = $bridge;
+    $this->trackingConfig = $trackingConfig;
+    $this->subscribersFeature = $subscribersFeature;
+  }
+  
+  public function validate(NewsletterEntity $newsletterEntity): ?string {
+    try {
+      $this->validateSubscriberLimit();
+      $this->validateBody($newsletterEntity);
+      $this->validateUnsubscribeRequirements($newsletterEntity);
+      $this->validateReEngagementRequirements($newsletterEntity);
+      $this->validateAutomaticLatestContentRequirements($newsletterEntity);
+    } catch (ValidationException $exception) {
+      return __($exception->getMessage(), 'mailpoet');
+    }
+    return null;
+  }
+
+  private function validateUnsubscribeRequirements(NewsletterEntity $newsletterEntity): void {
+    if (!$this->bridge->isMailpoetSendingServiceEnabled()) {
+      return;
+    }
+    $content = $newsletterEntity->getContent();
+    $hasUnsubscribeUrl = strpos($content, '[link:subscription_unsubscribe_url]') !== false;
+    $hasUnsubscribeLink = strpos($content, '[link:subscription_unsubscribe]') !== false;
+
+    if (!$hasUnsubscribeLink && !$hasUnsubscribeUrl) {
+      throw new ValidationException('All emails must include an "Unsubscribe" link. Add a footer widget to your email to continue.');
+    }
+  }
+
+  private function validateBody(NewsletterEntity $newsletterEntity): void {
+    $emptyBodyErrorMessage = 'Poet, please add prose to your masterpiece before you send it to your followers.';
+    $content = $newsletterEntity->getContent();
+
+    if ($content === '') {
+      throw new ValidationException($emptyBodyErrorMessage);
+    }
+
+    $contentBlocks = $newsletterEntity->getBody()['content']['blocks'] ?? [];
+    if (count($contentBlocks) < 1) {
+      throw new ValidationException($emptyBodyErrorMessage);
+    }
+  }
+
+  private function validateReEngagementRequirements(NewsletterEntity $newsletterEntity): void {
+    if ($newsletterEntity->getType() !== NewsletterEntity::TYPE_RE_ENGAGEMENT) {
+      return;
+    }
+
+    if (strpos($newsletterEntity->getContent(), '[link:subscription_re_engage_url]') === false) {
+      throw new ValidationException('A re-engagement email must include a link with [link:subscription_re_engage_url] shortcode.');
+    }
+
+    if (!$this->trackingConfig->isEmailTrackingEnabled()) {
+      throw new ValidationException('Re-engagement emails are disabled because open and click tracking is disabled in MailPoet → Settings → Advanced.');
+    }
+  }
+
+  private function validateAutomaticLatestContentRequirements(NewsletterEntity $newsletterEntity) {
+    if ($newsletterEntity->getType() !== NewsletterEntity::TYPE_NOTIFICATION) {
+      return;
+    }
+    $content = $newsletterEntity->getContent();
+    if (
+      strpos($content, '"type":"automatedLatestContent"') === false && 
```

Comment:
I'm not familiar with this part of the code, but I found this file in case it helps:

https://github.com/mailpoet/mailpoet/blob/c50715698ce7137b72003fe742682ced355b5a16/mailpoet/assets/js/src/newsletter_editor/blocks/automatedLatestContent.js

I also noticed that we seem to have this type of block defined in several of our templates. Examples:

https://github.com/mailpoet/mailpoet/blob/9f790efbf0f032fc32e264dc33450286b362976e/mailpoet/lib/Config/PopulatorData/Templates/Avocado.php#L712
https://github.com/mailpoet/mailpoet/blob/9f790efbf0f032fc32e264dc33450286b362976e/mailpoet/lib/Config/PopulatorData/Templates/Charity.php#L824

That being said, I only skimmed through what I found, so I might be missing something.

Reply:
Thanks for tracking that down, not sure how I missed it myself!  I ended up trying to test this myself by creating a newsletter with an ALC block, and in that case what I see generated for the body in the DB looks like this:

```json
      {
        "type": "automatedLatestContentLayout",
        "withLayout": true,
        "amount": "5",
        "contentType": "post",
        "terms": [],
        "inclusionType": "include",
        "displayType": "excerpt",
        "titleFormat": "h1",
        "titleAlignment": "left",
        "titleIsLink": false,
        "imageFullWidth": false,
        "titlePosition": "abovePost",
        "featuredImagePosition": "alternate",
        "fullPostFeaturedImagePosition": "left",
        "showAuthor": "no",
...
```

I'm guessing perhaps there's a display option that changes the type from this to the version without `Layout`. In any case, there's no need I think to dig any further into this. Having the check for both definitely seems like what we want.